engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -528,26 +522,20 @@ object ProgramExecutionSupport {
     * @return either a byte array representing the visualization result or an
     *         error
     */
-  @scala.annotation.tailrec
   private def visualizationResultToBytes(
     value: AnyRef
-  ): Either[VisualisationException, Array[Byte]] =
-    value match {
-      case text: String =>
-        Right(text.getBytes(StandardCharsets.UTF_8))
-      case text: Text =>
-        Right(text.toString.getBytes(StandardCharsets.UTF_8))
-      case bytes: Array[Byte] =>
-        Right(bytes)
-      case withWarnings: WithWarnings =>
-        visualizationResultToBytes(withWarnings.getValue)
-      case other =>
-        Left(
-          new VisualisationException(
-            s"Cannot encode ${other.getClass} to byte array."
-          )
+  ): Either[VisualisationException, Array[Byte]] = {
+    val bytes = VisualizationResult.visualizationResultToBytes(value)
+    if (bytes != null) {
+      Right(bytes)
+    } else {
+      Left(
+        new VisualisationException(
+          s"Cannot encode ${value.getClass} to byte array."
         )
+      )
     }
```

Comment:
```suggestion
    Option(VisualizationResult.visualizationResultToBytes(value))
      .toRight(
        new VisualisationException(
          s"Cannot encode ${value.getClass} to byte array."
        )
      )
```

Reply:
OK, done in [7e44537](https://github.com/enso-org/enso/pull/4045/commits/7e445373b52864eac0bb2794bb5ed4541c7131b6)